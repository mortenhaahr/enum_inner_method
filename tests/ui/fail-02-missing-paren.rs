use enum_inner_method::enum_inner_method;

pub struct Circle;
impl Circle {
    pub fn area(&self) -> f64 {
        3.14
    }
}

pub struct Rectangle;
impl Rectangle {
    pub fn area(&self) -> f64 {
        6.0
    }
}

#[enum_inner_method(area -> f64)]
pub enum ShapeEnum {
    Circle(Circle),
    Rectangle(Rectangle),
}

fn main() {
}
