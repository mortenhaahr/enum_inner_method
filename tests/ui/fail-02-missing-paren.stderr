error: expected parentheses
  --> tests/ui/fail-02-missing-paren.rs:17:26
   |
17 | #[enum_inner_method(area -> f64)]
   |                          ^
